src/Keboola/S3Extractor/ConfigDefinition.php

```diff
@@ -57,4 +69,29 @@ protected function getParametersDefinition(): ArrayNodeDefinition
        // @formatter:on
         return $parametersNode;
     }
+
+    private function addValidate(NodeDefinition $definition): void
+    {
+        $definition->validate()->always(function ($item) {
+            if (!isset($item['loginType']) || $item['loginType'] === 'credentials') {
```

Comment:
Ak je tam default value (a prida sa tam aj ta `enumNode`) ... tak `!isset($item['loginType']) === true` nie je platny stav ... a teda by to malo byt pridane IMHO nakoniec do `else` .... `throw new LogicException()` (alebo nieco podobne).

Reply:
ok je to tam